### ISSUE TITLE
Uncaught Reference error:  undefined is not a function

### ISSUE THREAD

**Issue by sidd-shah** [asaglimbeni/django-datetime-widget#52]
I have been trying to get the widget running but the it keeps giving the above error. Please help.
Form:

class BestQuoteQueryForm(forms.ModelForm):
    class Meta:
        model = BestQuoteQuery
        fields = ('email', 'name', 'number', 'additional_requirement','enquiry_for_date','event_type','no_of_guests','event_timing_start','event_timing_end')
        widgets = {
            'event_timing_start':TimeWidget(),
            'event_timing_end':TimeWidget(),
        }

HTML

<div class="form-group">
            <div><label>Occasion Time</label></div>

```
   <div id="id_event_timing_start"  class="controls input-append date">
       <input id="id_event_timing_start" name="event_timing_start" readonly="" type="text" />
       <span class="add-on"><i class="icon-remove"></i></span>
       <span class="add-on"><i class="icon-th"></i></span>
   </div>
   <script type="text/javascript">
       $("#id_event_timing_start").datetimepicker({minView: 0,
```

maxView: 1,
autoclose: true,
startView: 1,
format: 'hh:ii'});
       </script>
       <div id="id_event_timing_end"  class="controls input-append date">
           <input class="form-control" id="id_event_timing_end" name="event_timing_end" readonly="" type="text" />
           <span class="add-on"><i class="icon-remove"></i></span>
           <span class="add-on"><i class="icon-th"></i></span>
       </div>
       <script type="text/javascript">
           $("#id_event_timing_end").datetimepicker({minView: 0,
maxView: 1,
autoclose: true,
startView: 1,
format: 'hh:ii'});
       </script>
        </div>
<scri[t src="http://ajax.googleapis.com/ajax/libs/jquery/2.0.0/jquery.min.js"></script>

<script src="/static/scripts/jquery.min.js"></script>

<script src="/static/scripts/bootstrap.min.js"></script>"

**Comment by rtp-calpoly** [comment]
I am having exactly the same problem. I have just installted the app within my Django project and, when I try to load the page, I get the same error:

** website/settings.py

    INSTALLED_APPS = (
        'django.contrib.auth',
        'django.contrib.contenttypes',
        'django.contrib.sessions',
        'django.contrib.sites',
        'django.contrib.messages',
        'django.contrib.staticfiles',

        'allauth',
        'allauth.account',
        'session_security',
        'leaflet',
        'periodically',
        'rpc4django',
        'rest_framework',
        'django_extensions',
        'south',
        'datetimewidget',

        # ... and some more that I do not list       

    )

** forms.py:

    from django import forms as django_forms
    from datetimewidget import widgets as datetime_widgets
    from services.leop.models import launch as launch_models

    class LaunchForm(django_forms.ModelForm):
        class Meta:
            model = launch_models.Launch
            fields = ('identifier', 'date')
            widgets = {
                'date': datetime_widgets.DateTimeWidget(
                    usel10n=True, bootstrap_version=3
                )
            }

The error that I am getting is a Javascript error: when jQuery tries to create the component, it cannot find the function (take a look at the attached screenshot).

![screenshot from 2015-01-10 19 50 57](https://cloud.githubusercontent.com/assets/5666358/5694280/56f11a5e-9901-11e4-842f-bb482c385ecb.png)

What am I doing wrong?

**Comment by ollytheninja** [comment]
If it's the jquery datetimepicker() function it can't find I suspect the following isn't being included on the page anywhere.
`<link href="/site_media/static/css/datetimepicker.css" type="text/css" media="all" rel="stylesheet" />`
`<script type="text/javascript" src="/site_media/static/js/bootstrap-datetimepicker.js"></script>`

**Comment by rtp-calpoly** [comment]
Yes, you are right. It has not been included anywhere in the website since in the installation section of the website doesn't say that that js and css file has to be included... sorry if I was supposed to know that beforehand...

**Comment by ollytheninja** [comment]
Step 3 of [basic configuration](https://github.com/asaglimbeni/django-datetime-widget#basic-configuration) does say it but it is easy to miss - I had the same problem earlier today :)
I think the reason it is in configuration and installation is that you should really only include it on pages where you are actually using the widget.

**Comment by rtp-calpoly** [comment]
Actually, if you read carefully the HTML code, at least in my case, it looks like that code is meant to include "bootstrap" dependencies and not "bootstrap-datetimepicker"... that was the problem that I had: I read the section and I thought that I only would have to include bootstrap js/css files... it was weird at first, but I decided to stick to the manual... please correct me if I am wrong.

**Comment by ollytheninja** [comment]
Sorry, yes you're right, you shouldn't have to include the css/js directly as long as you include `{{ form.media }}` somewhere in your template.
I was probably having problems because I was using the widgets without a form and I just include
`{% if form %}{{form.media}}{% endif %}`
in my base template.
